crypto/ec/curve25519.c

```diff
@@ -5448,6 +5489,17 @@ ossl_ed25519_sign(uint8_t *out_sig, const uint8_t *message, size_t message_len,
     unsigned int sz;
     int res = 0;
 
+    if (context == NULL)
+        context_len = 0;
+
+    /* if csflag is set, then a non-empty context-string is required */
+    if (csflag && context_len == 0)
+        goto err;
```

Comment:
the first 2 gotos could be changed to return 0 if you do the fetch and new later..

Reply:
Agreed, but I don't think that change offers much benefit.  I prefer to leave the code as is.

Reply:
See chapter 7 of https://www.openssl.org/policies/technical/coding-style.html